haystack/components/converters/openapi_functions.py

```diff
@@ -176,22 +216,26 @@ def _parse_openapi_spec(self, content: str) -> Dict[str, Any]:
         :return: The parsed OpenAPI specification.
         :rtype: Dict[str, Any]
         """
+        open_api_spec_content = None
         try:
-            return json.loads(content)
+            open_api_spec_content = json.loads(content)
```

Comment:
We want to run `return jsonref.replace_refs(open_api_spec_content)` here next otherwise the next try block gets executed.

Reply:
Very good catch :-)